plugins/inputs/intel_powerstat/fetcher.go

```diff
@@ -0,0 +1,139 @@
+//go:build linux && amd64
+
+package intel_powerstat
+
+import (
+	ptel "github.com/intel/powertelemetry"
+)
```

Comment:
The upstream library provides these exported methods. I thought that having this interfaces in library and returning `MetricFetcher` from `powertelemetry.New` would cause backward incompatible changes if any new methods are added to  these interfaces. I decided to define accepting interfaces, from plugin side instead.

Additionally this library will be used as a dependency for other repos, I did not want to add these interfaces there as other projects won't need all methods in these definitions. This gives more flexibility to the consumer to decide what is going to be used, and implemented also for testing purposes. let me know what do you think.

Reply:
To be honest I don't even see why you want to hide this behind an interface... Usually you use an interface if you can envision multiple different implementations, which doesn't seem to be the case here... 

> I thought that having this interfaces in library and returning `MetricFetcher` from `powertelemetry.New` would cause backward incompatible changes if any new methods are added to these interfaces.

Can you elaborate on this? If you add a new function to the interface, the plugin will continue to work. The only issue is if you remove a function. This being said, I would remove those interfaces completely and use the actual implementation type `*ptel.PowerTelemetry` directly.